From e7188f07768779557ade8bcc53b5ce6bf7351caa Mon Sep 17 00:00:00 2001
From: Anas Karbila <anaskarbila@gmail.com>
Date: Sun, 16 Apr 2017 21:50:12 +0900
Subject: [PATCH 32/67] Gallery: TileImageView: fix NPE

* GetGLRoot() can return null when the job is still running but the GL has already been detached

Change-Id: Ic58b4752f64390efdf14ca1a33a81a8f77100ee0
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 src/com/android/gallery3d/ui/TileImageView.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/gallery3d/ui/TileImageView.java b/src/com/android/gallery3d/ui/TileImageView.java
index dd4b406fd..5edd9d7f0 100644
--- a/src/com/android/gallery3d/ui/TileImageView.java
+++ b/src/com/android/gallery3d/ui/TileImageView.java
@@ -476,7 +476,7 @@ public class TileImageView extends GLView {
         synchronized (this) {
             mUploadQueue.push(tile);
         }
-        if (mTileUploader.mActive.compareAndSet(false, true)) {
+        if (getGLRoot() != null && mTileUploader.mActive.compareAndSet(false, true)) {
             getGLRoot().addOnGLIdleListener(mTileUploader);
         }
     }
-- 
2.17.1

